mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -142,6 +138,22 @@ def cleanup(self):
         self.normaliseStackSelector.unsubscribe_from_main_window()
         self.main_window.spectrum_viewer = None
 
+    def on_visibility_change(self) -> None:
+        """
+        When the visibility of an ROI is changed, update the visibility of the ROI in the spectrum widget
+        """
+        for roi_item in range(self.roi_table_model.rowCount()):
+            if self.presenter.export_mode == ExportMode.ROI_MODE:
+                if self.roi_table_model.row_data(roi_item)[2] is False:
+                    self.set_roi_alpha(0, roi_item)
+                else:
+                    self.set_roi_alpha(255, roi_item)
+                    self.presenter.redraw_spectrum(self.roi_table_model.row_data(roi_item)[0])
+            else:
+                self.set_roi_alpha(0, roi_item)
+
+        return
```

Comment:
NITPICK, although the efficiency gain is likely negligible, we could reduce the number of calls to `self.roi_table_model.row_data(roi_item)` in the loop by instead placing the row data in a variable (`row_data = self.roi_table_model.row_data(roi_item)`) before the loop to be used within the loop.

Reply:
It still needs to be in the loop, because `roi_item` get incremented. But I can be done once and put in names to improve readability.

Reply:
Yes, that's correct, apologies, I didn't word this very well. 
I was thinking something like this:
```Python
def on_visibility_change(self) -> None:
    """
    When the visibility of an ROI is changed, update the visibility of the ROI in the spectrum widget
    """
    for roi_item in range(self.roi_table_model.rowCount()):
        row_data = self.roi_table_model.row_data(roi_item)
        if self.presenter.export_mode == ExportMode.ROI_MODE:
            if row_data[2] is False:
                self.set_roi_alpha(0, roi_item)
            else:
                self.set_roi_alpha(255, roi_item)
                self.presenter.redraw_spectrum(row_data[0])
        else:
            self.set_roi_alpha(0, roi_item)
```